src/internal/connector/exchange/service_iterators.go

```diff
@@ -95,18 +95,18 @@ func filterContainersAndFillCollections(
 
 		added, removed, newDelta, err := getter.GetAddedAndRemovedItemIDs(ctx, qp.ResourceOwner, cID, prevDelta)
 		if err != nil {
+			// note == nil check; only catches non-inFlight error cases.
```

Comment:
not sure I understand this comment. Given we got an error doesn't that mean the request was inflight? Maybe we're thinking of different spans of time?

Reply:
eh, it's a semantics thing.  `IsErrDeletedInFlight` will return nil if the error doesn't match the sentinel.  This is because the call may wrap the error if it isn't the expected underlying struct, for future errors.As checks.  I might revisit this, as it is confusing (like you're pointing out), but it's sufficient for now.